### ISSUE TITLE
Generated Layouts are predicted outside of canvas

### ISSUE THREAD

**Issue by Sagnnik** [JulianGuerreiro/LayoutFlow#6]
I have trained the model on Posterlayout dataset and the generated bounding boxes are predicted outside of the canvas. I am not sure why this is the case. Maybe the model isn't able to figure out the data distribution. The training seems to be fine. I checked the samples over the training steps they are alright but the generated samples are the issue.

Sample generated output (Normalised) (format = 'xywh'):
tensor([[0.4872, 0.0679, 0.4561, 0.0920],
        [0.5072, 0.2176, 0.7271, 0.1173],
        [0.4966, 0.3397, 0.7602, 0.0667],
        [0.4920, 0.3318, 0.6628, 0.0387]])

here is the [link](https://wandb.ai/sagnnikbiswas2002/LayoutFlow/runs/u7rc5rcp?nw=nwusersagnnikbiswas2002) to the run on wandb

**Comment by JulianGuerreiro** [comment]
Hi,
thank you for your message. Given the information that you provided, I don't see what could have gone wrong. It seems that the Overlap and Alignment values are very high during training. In particular, the Overlap value is increasing during training, so it seems that something is off. I would maybe try adjusting the learning rate or other hyper-parameters. I would also recommend using the visualization to check whether the layout is actually improving during training, both the training loss and validation loss are usually not that reliable to assess the real generation quality.

**Comment by Sagnnik** [comment]
So, the issue was in the labels of the dataset. you were using labels [1,2,3, ...] (starting from 1 basically), while the dataset used [0,1,2] ([link](https://huggingface.co/datasets/creative-graphic-design/PKU-PosterLayout?row=0)to the dataset). Even then it generating outside of the canvas like the sample I have written previously. But if I used visualization.py code that you have provided the final layout turns out fine. I am confused about that part otherwise the model is working fine 

P.S. - this is my final [run](https://wandb.ai/sagnnikbiswas2002/LayoutFlow/runs/jbb0hyop?nw=nwusersagnnikbiswas2002) on posterlayout. I have used the same fid model as provided in the dataset but I haven't finetuned it on the newer labels (starting from 1 instead of 0). It is giving me decent results so I left it as it is.

**Comment by JulianGuerreiro** [comment]
Hi, thank you for the update! I'm glad that it seems to be working now. The reason why the zero label is, is because it acts as the padding label. Since layouts can have a variable number of elements, elements that are not should not be considered are labeled with zero. 
Regarding the elements being outside of the canvas, it seems that it is fine when using the visualization script, so maybe the values that you looked at still required some additional formatting.
